central/plugins/org.jboss.tools.central/src/org/jboss/tools/central/installation/CentralInstallationPage.java

```diff
@@ -45,49 +48,59 @@ public void createControl(Composite parent) {
 		layoutData.widthHint = 700;
 		earlyAccessLabel.setLayoutData(layoutData);
 		
-		FilteredTree iusViewer = new FilteredTree(parent, SWT.MULTI | SWT.FULL_SELECTION | SWT.H_SCROLL | SWT.V_SCROLL | SWT.BORDER, new PatternFilter(), false);
-		IUColumnConfig[] columnsConfig = new IUColumnConfig[] {new IUColumnConfig(ProvUIMessages.ProvUI_NameColumnTitle, IUColumnConfig.COLUMN_NAME, ILayoutConstants.DEFAULT_PRIMARY_COLUMN_WIDTH), new IUColumnConfig(ProvUIMessages.ProvUI_VersionColumnTitle, IUColumnConfig.COLUMN_VERSION, ILayoutConstants.DEFAULT_SMALL_COLUMN_WIDTH), new IUColumnConfig(ProvUIMessages.ProvUI_IdColumnTitle, IUColumnConfig.COLUMN_ID, ILayoutConstants.DEFAULT_COLUMN_WIDTH), new IUColumnConfig(ProvUIMessages.ProvUI_ProviderColumnTitle, IUColumnConfig.COLUMN_PROVIDER, ILayoutConstants.DEFAULT_COLUMN_WIDTH)};
-		iusViewer.getViewer().setLabelProvider(new IUDetailsLabelProvider(iusViewer, columnsConfig, parent.getShell()));
-		// copied from AvailableIUGroup
-		Tree tree = iusViewer.getViewer().getTree();
-		tree.setHeaderVisible(true);
-		for (IUColumnConfig column : columnsConfig) {
-			TreeColumn tc = new TreeColumn(tree, SWT.NONE);
-			tc.setResizable(true);
-			tc.setText(column.getColumnTitle());
-			tc.setWidth(column.getWidthInPixels(tree));
-		}
-		iusViewer.getViewer().setContentProvider(new ITreeContentProvider() {
-			@Override
-			public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
-			}
-			
-			@Override
-			public void dispose() {
-			}
-			
-			@Override
-			public boolean hasChildren(Object element) {
-				return false;
-			}
-			
-			@Override
-			public Object getParent(Object element) {
-				return null;
+		InstallationChecker installChecker = null;
+		try {
+			installChecker = InstallationChecker.getInstance();
```

Comment:
so InstallationChecker.getInstance() is called in the UI thread here, but inside a Job in JBossCentralEditor. Is it supposed to be long running or?